src/Rings/mpoly-graded.jl

```diff
@@ -1637,31 +1819,65 @@ ideal(x*y - z^2, -x^3 + x^2*z)
 
 julia> dehomogenization(I, 3)
 ideal(x*y - 1, -x^3 + x^2)
+
+julia> W = [1 2 1 0; 3 4 0 1]
+2×4 Matrix{Int64}:
+ 1  2  1  0
+ 3  4  0  1
+
+julia> S, (w, x, y, z) = GradedPolynomialRing(QQ, ["w", "x", "y", "z"], W)
+(Multivariate Polynomial Ring in w, x, y, z over Rational Field graded by
+  w -> [1 3]
+  x -> [2 4]
+  y -> [1 0]
+  z -> [0 1], MPolyElem_dec{fmpq, fmpq_mpoly}[w, x, y, z])
+
+julia> F = w^3*y^3*z^3 + w^2*x*y^2*z^2 + w*x^2*x*z + x^3
+w^3*y^3*z^3 + w^2*x*y^2*z^2 + w*x^3*z + x^3
+
+julia> dehomogenization(F, 3)
+w^3 + w^2*x + w*x^3 + x^3
 ```
 """
 function dehomogenization(F::MPolyElem_dec, pos::Int)
   S = parent(F)
-  A = String.(symbols(S))
-  l = length(A)
-  if (pos > l+1) ||  (pos <1)
+  @assert is_zm_graded(S)
+  l = ngens(S)
+  G = grading_group(S)
+  m = ngens(G)
+  if (pos > l-m+1) ||  (pos <1)
       throw(ArgumentError("Index out of range."))
   end
-  deleteat!(A, pos)
+  for i = 1:m
+      @assert degree(gens(S)[pos-1+i]) == gen(G, i)
+  end
+  A = String.(symbols(S))
```

Comment:
If all code accepted strings *and* symbol, then you'd not have to make this conversion